Update about automattic library to latest v0.0.4

ravishanker opened the pull request:
Updated about automattic library version from 0.0.2 to 0.0.4

This PR updates the about-automattic library version v0.0.2 to v0.0.4 and includes changes from [0.0.3](https://github.com/Automattic/about-automattic-android/releases/tag/0.0.3) and [0.0.4](https://github.com/Automattic/about-automattic-android/releases/tag/0.0.4)

Fixes 30-gh-Automattic/about-automattic-android

To test:

- Use a Samsung device like s10e or z-flip if you have one, otherwise create an emulator with an image like 6.7 Horizontal Fold-in from AVD Manager in AS
- Launch app, On MySite, Go to Me by clicking on avatar on top right hand corner
- Notice About WordPress (second from bottom), click on it to see new Unified About 
- Notice tumbler logo is not cut-off and fits withing the width


## Regression Notes
1. Potential unintended areas of impact
None that is known

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing


3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Update about automattic library to latest v0.0.4

Updated about automattic library version from 0.0.2 to 0.0.4

ravishanker committed:
Update strings.xml

Removed unused strings, they are already moved to library

AliSoftware commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -3864,22 +3864,8 @@ translators: %s: Select control option value e.g: "Auto, 25%". -->
     <string name="about_automattic_legal_page_title">Legal and More</string>
     <string name="about_automattic_acknowledgements_page_title">Acknowledgements</string>
     <string name="about_automattic_version_label">Version %1$s</string>
-    <string name="about_automattic_share_with_friends">Share with friends</string>
-    <string name="about_automattic_rate_us">Rate us</string>
-    <string name="about_automattic_instagram">Instagram</string>
-    <string name="about_automattic_twitter">Twitter</string>
-    <string name="about_automattic_legal_and_more">Legal and more</string>
-    <string name="about_automattic_family">Automattic family</string>
-    <string name="about_automattic_work_with_us">Work with us</string>
-    <string name="about_automattic_work_from_anywhere">Work from anywhere</string>
-    <string name="about_automattic_terms_of_service">Terms of service</string>
-    <string name="about_automattic_privacy_policy">Privacy policy</string>
-    <string name="about_automattic_california_privacy_notice">California privacy notice</string>
-    <string name="about_automattic_source_code">Source code</string>
-    <string name="about_automattic_acknowledgements">Acknowledgements</string>
```
@ravishanker Are we sure about those entries being deleted?

Did they stop being needed starting 0.0.3 or 0.0.4 of the lib? Feels suspicious to me 🤔

AliSoftware replied:
Actually since these entries are copied from [the library's own `strings.xml` file](https://github.com/Automattic/about-automattic-android/blob/0.0.4/library/src/main/res/values/strings.xml) by our release scripts, so that they can be properly imported into GlotPress alongside the app strings, those should actually not be modified manually in PRs and are to still be in the app's `strings.xml` for that reason (at least until we make localization of libraries to be independant to their client apps, see pxLjZ-6Fc-p2).

So this change to the strings needs to be reverted.

AliSoftware replied:
After further investigation: those indeed need to be removed, but NOT because they're now "moved to the library" as the commit implied, but rather because the keys got renamed in the lib (as part of https://github.com/Automattic/about-automattic-android/commit/ea6045821722c67b4235fd7121d5ac619dac4a1c) during 0.0.4 and the old key names are unused (even in the library itself).

AliSoftware reviewed (commented):
Tested the Instsallable Build on Samsung Galaxy S20 and works as expected 👍 

<img alt="About Screen on Samsung S20" src="https://user-images.githubusercontent.com/216089/148752157-9773c86b-bfd9-4e0a-bdc9-3d282ae34730.jpg" width="50%" />

Not approving though because I'm suspicious about the English strings being removed from `strings.xml`. Can you double-check about those?

AliSoftware committed:
Revert "Update strings.xml"

This reverts commit 4c3b085742eec1aa8509bfd2f0aa5bed030af2bc.

The strings from libraries should not be removed or edited from the app's `strings.xml`. Those are needed to be in the app's own `strings.xml` because of our translation pipeline requires it. They are copied over from the libs' `strings.xml` into the app's one during code freeze and should remain there in order to be translated as expected. This is a glitch of our translation pipeline that we hope to address at some point (see pxLjZ-6Fc-p2).

AliSoftware reviewed (approved):
Decided to fix the PR myself via 8a83dc5219aca2119866e0fca11f8608f5ef85aa so that I can include it in 19.0 before doing the code-freeze.

ParaskP7 reviewed (approved):
👋 @ravishanker / @AliSoftware !

I have reviewed and tested this PR as per the instructions, everything works as expected, good job! 🌟 

Tested on `6.7 Horizontal Fold-in` emulator.

AliSoftware committed:
Restore "Update strings.xml"

This reverts commit 8a83dc5219aca2119866e0fca11f8608f5ef85aa and restores 4c3b085742eec1aa8509bfd2f0aa5bed030af2bc

The reason initially stated for 4c3b085742eec1aa8509bfd2f0aa5bed030af2bc was misleading, as it stated those strings had to be removed “because they were moved into the library since”.
But our translation pipeline still requires the strings used by our internal binary dependencies (GBM, Login, Stories, About libs) to be merged into the app’s main `strings.xml` (which our release-toolkit takes care of during code-freeze), in order for those strings to be imported into GlotPress and translated.

After investigating further, those strings still need to be removed indeed (which is why this commit restores their removal after all), but NOT because “they were moved into the library”, and instead because those strings keys don’t exist in the 0.0.4 version of the library anymore, and got renamed to something else (mostly adding the `*_item_title` suffix to most of them, see https://github.com/Automattic/about-automattic-android/commit/ea6045821722c67b4235fd7121d5ac619dac4a1c). This is the real reason why those old and now unused keys need to be removed.

Note that during next code freeze, the new names of those keys (with `*_item_title`) will be added to the `strings.xml` of the app to get those new names translated via GlotPress (even if in practice the translators will get the translated copy from the old keys suggested to them from the translation memory, making it easy for them to use the translation from the old keys as the one to use for the renamed keys)

AliSoftware commented:
Ok. after much investigation on this issue on the `strings.xml` file, I've written the results of my investigation in the commit message of 5179232, which restores their deletion.

The TL;DR is that those string keys indeed need to be removed, but not for the reason mentioned by Ravi in his initial commit ("because they were moved into the library"), but instead because those old keys [don't exist and are not used by the library anymore in its 0.0.4 version](https://github.com/Automattic/about-automattic-android/commit/ea6045821722c67b4235fd7121d5ac619dac4a1c) (and were renamed since). This is the real reason why the linter (rightfully) complains about them being unused (even by the lib itself).

Since the keys got renamed in the library, one solution to solve the linter errors would have been to update all the relevant/renamed keys in the `values-*/strings.xml` with their new key name instead. But in practice, the copy of the strings used by the library into the app's `values/strings.xml` (so that they get included when imported by GlotPress for translation) is handled by our automated release scripts during code-freeze. So the right solution is indeed to remove the keys that are not used by the library anymore (5179232), and later let our scripts add the new (renamed) strings keys to the main `values/strings.xml` file during the next code-freeze.
